src/tiled/tileseteditor.cpp

```diff
@@ -235,6 +241,7 @@ void TilesetEditor::removeDocument(Document *document)
     tilesetDocument->disconnect(this);
 
     TilesetView *view = mViewForTileset.take(tilesetDocument);
+    Preferences::instance()->setTerrainEditorZoom(tilesetDocument->fileName(),view->scale());
```

Comment:
Please call it `setTilesetScale`, since this isn't only for terrain and the property is already called `scale`.

Coding style: please put a space after `,`